docusaurus/docs/React/custom-code-examples/channel-header.mdx

```diff
@@ -0,0 +1,174 @@
+---
+id: channel_header
+sidebar_position: 9
+title: ChannelHeader
+---
+
+import CustomChannelHeader from '../assets/CustomChannelHeader.png';
+
+In this example, we will demonstrate how to use a custom `ChannelHeader` component instead of the default. Unlike most of the other library components where a prop on a parent component will override the default, for this component all you need to do is use
+your custom component in place of the library's header option.
+
+### Implementation
+
+The default header displays the name of the channel, the number of members, and the number of members online. In our example, we will display the name and add the typing indicator below. We need to pull the `channel` information from context.
+
+:::note
+Since `channel` is drawn from the `ChannelStateContext`, we must create an inner component that is a child of `Channel` and
+call the `useChannelStateContext` custom hook.
+:::
+
+```jsx
+const CustomChannelHeader = (props: ChannelHeaderProps) => {
+    const { title } = props;
+
+    const { channel } = useChannelStateContext();
+    const { name } = channel.data || {};
+
+    return (
+        <div>{title || name}{' '}</div>
+    )
+};
+
+const ChannelInner = () => {
```

Comment:
we don't need the ChannelInner, can just replace this part with the ChannelInner component in the App return